### PR TITLE
[#2474]add-index-check

### PR DESCRIPTION
**Pull request by bwanglzu**
Fixes #
### Proposed fixes:

Fix issue [#2474](https://github.com/ckan/ckan/issues/2474), add primary key check before append index into indexlist
### Features:
- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ X] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

File: ckanext/datastore/db.py [hunk 1]
```diff
@@ -430,7 +430,8 @@ def create_indexes(context, data_dict):
 
     if primary_key is not None:
         _drop_indexes(context, data_dict, True)
-        indexes.append(primary_key)
+        if primary_key not in indexes:
+            indexes.append(primary_key)
```
Comment (wardi) [inline ckanext/datastore/db.py]: primary key and indexes might be given as strings or lists, so this won't catch the case of the same index being specified in different ways. Maybe move this fix below after the call to `datastore_helpers.get_list(index)`

**Comment by bwanglzu** [comment]
Hi, wardi. I'm a bit confused. 
1. Do you mean that both `primary_key` and `indexes` could be list?
2. If `primary_key` could be list, if I move the fix blow after `datastore_helpers.get_list(index) `, we can not ensure `primary_key` contains no duplicate elements, and we can only ensure current `primary_key` not in variable `index_fields`, rather than the whole possible indexes.

**Comment by wardi** [comment]
What I mean is that you could have: `indexes=['pka,pkb', 'another']` and `primary_key=['pka', 'pkb']` and you'll still end up with a duplicate because the `datastore_helpers.get_list` normalization happens inside the loop below. Better to check for a repeated index inside that loop.

**Comment by kmbn** [comment]
We decided to close old pull requests that are not actively worked on so that we can focus our effort and attention on PRs made against the current version of CKAN. A lot has changed in the last several months and many PRs will have gone stale or may now be in conflict with the latest version.

Please note: this isn't a judgement about the quality of your pull request. We just don't have the resources to review older PRs.

If you would like to reopen your PR (and update it if necessary), please do!

**Pull request closed without merging**